logging/src/posix/logging.cc

```diff
@@ -6,7 +6,8 @@ using namespace std;
 
 namespace leatherman { namespace logging {
 
-    void colorize(ostream& dst, log_level level)
+    template <typename StreamT>
+    void colorize_impl(StreamT& strm, log_level level)
```

Comment:
I use that with [this CMake option](https://github.com/puppetlabs/pxp-agent/blob/master/CMakeLists.txt#L12} because I like colors when debugging (mostly other PCP components) while keeping logs on file for future reference.

The `boost::log::formatting_ostream` stream is provided directly to the color_formatter() handler.
The class, defined [here](https://github.com/boostorg/log/blob/master/include/boost/log/utility/formatting_ostream.hpp#L88), is used [here](https://github.com/boostorg/log/blob/master/include/boost/log/utility/formatting_ostream_fwd.hpp#L41) for the type:
```
#ifdef BOOST_LOG_USE_CHAR
typedef basic_formatting_ostream< char > formatting_ostream;
#endif
```